examples/bazel_simple/WORKSPACE

```diff
@@ -0,0 +1,53 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
+    name = "com_google_protobuf",
+    commit = "09745575a923640154bcf307fba8aedff47f240a",
+    remote = "https://github.com/protocolbuffers/protobuf",
+    shallow_since = "1558721209 -0700",
+)
+
+load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
+
+protobuf_deps()
+
+http_archive(
+    name = "rules_python",
+    sha256 = "aa96a691d3a8177f3215b14b0edc9641787abaaa30363a080165d06ab65e1161",
+    url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.1/rules_python-0.0.1.tar.gz",
+)
+
+load("@rules_python//python:repositories.bzl", "py_repositories")
+
+py_repositories()
+
+load("@rules_python//python:pip.bzl", "pip_import", "pip_repositories")
+
+pip_repositories()
+
+http_archive(
+    name = "build_stack_rules_proto",
+    sha256 = "35779b6eadae57f58e65e717e8fcb2bc19408977ebb48412d6eff708048e0fb5",
+    strip_prefix = "rules_proto-95ea73aef50fcf04e91f379589fe37be29073be2",
+    urls = ["https://github.com/reflexe/rules_proto/archive/95ea73aef50fcf04e91f379589fe37be29073be2.tar.gz"],
+)
+
+load("@build_stack_rules_proto//python:deps.bzl", "python_proto_library")
+
+python_proto_library()
+
+pip_import(
+    name = "protobuf_py_deps",
+    requirements = "@build_stack_rules_proto//python/requirements:protobuf.txt",
+)
+
+load("@protobuf_py_deps//:requirements.bzl", protobuf_pip_install = "pip_install")
+
+protobuf_pip_install()
+
+git_repository(
+    name = "com_github_nanopb_nanopb",
+    commit = "766ee1cebf6e1500dd9bd94fc217abb554ef55e7",
```

Comment:
What should be done with the git hash here? Would it be possible to use the already checked out version of nanopb by default, and have a commented example of referring to a particular git hash? That would make automated testing of the build rules a bit easier.

Reply:
In general, if it is an example the commit should remain constant. If it is part of documenation, then it is usually idiomatic to have this either as a TODO (e.g.`commit = "TODO",`) or alternatively it should reference the latest release. The motivation behind making using the git commit's in bazel is so that your dependencies are deterministic regardless of when you fetch it. Example idiomatic documentation from [rules_python](https://github.com/bazelbuild/rules_python).

I would happily include an automated test target, however this would likely look a little different.
```py
local_repository(
    name = "com_github_nanopb_nanopb",
    path = "../..",
)
The difference here is is references a path on your current file system in contrast to a remote repository. So it wouldn't require getting the latest commit.

Reply:
Hmm, but when people check out the nanopb repository and build the example, wouldn't they expect to build it against the revision they've just checked out?

As for testing, I like to have automatic builds of the examples exactly like users would build them, as otherwise it is quite easy to break them without noticing. Typically that's just a matter of adding a command to a Jenkins job and checking if there are any build errors. But if the example is building always against some old revision, it gives no information on whether the build rules still work with the newest version.